src/Domain/Encryption/Crypter.spec.ts

```diff
@@ -0,0 +1,62 @@
+import 'reflect-metadata'
+
+import { Crypter } from './Crypter'
+
+describe('Crypter', () => {
+  const encryptionIterations = 1000
+  const userServerKey = 'my-private-key'
+
+  const createCrypter = () => new Crypter(encryptionIterations)
+
+  it('should encrypt and decrypt data', async () => {
+    const sampleData = 'My-Super-Secret-Data'
+
+    const encryptedSampleData = await createCrypter().encrypt(Crypter.ENCRYPTION_VERSION_1, sampleData, userServerKey)
+
+    expect(encryptedSampleData).toHaveLength(156)
+
+    expect(
+      await createCrypter().decrypt(Crypter.ENCRYPTION_VERSION_1, encryptedSampleData, userServerKey)
+    ).toEqual('My-Super-Secret-Data')
```

Comment:
Should use `sampleData` var here. `.toEqual(sampleData)`

Reply:
not always but a good practice is to not reuse instantiated before variables as they may be subject of modifications within a call. This way we know that the outcome is exactly what we expect

Reply:
True, would seem to make sense for mutable data structures.